packages/headless/src/controllers/core/facets/category-facet/headless-core-category-facet.ts

```diff
@@ -141,10 +141,33 @@ export interface CoreCategoryFacetState {
   /** The facet ID. */
   facetId: string;
 
-  /** The facet's parent values. */
+  /**
+   * The facet's values, represented in their hierarchical form
+   * if there's an hierarchy, flat otherwise.
+   */
+  valuesAsTrees: CategoryFacetValue[];
+
+  /**
+   * The facet's selected value if any, undefined otherwise.
+   */
+  activeValue: CategoryFacetValue | undefined;
```

Comment:
should this be:
```suggestion
  activeValue?: CategoryFacetValue;
```

Reply:
Hmm ts playground inc to indicate why no

Reply:
https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgPIAczAPYjgG2QG8AoZZdKbdAfgC5kBnMKUAcwG4SBfEk0SLEQpUUAKogAJhBigIk4mQpV0DZqxBsAPgFcpMuZJ59pCfHCgoEuZsmpZcBBhgd58JU+cvJrIW9nF9WRB5Z0DpYPk+exw3JjgsRlkIRjRwgxCFEgCJCMN4xOTUl1iCIA

If you have `prop: Type | undefined` in an interface, you guarantee that the key `prop` is present in an object that respect the interface
If you have `prop?: type` however, you do not offer this guarantee.

Reply:
yeah I know that the union type means the key is required, my question is more so about whether we should have the key be present when there's no selected value. It's not something I feel strongly about though.